src/frontend/src/pages/MainPage/pages/filesPage/index.tsx

```diff
@@ -0,0 +1,340 @@
+      filter: "agTextColumnFilter",
+      cellClass: "cursor-text select-text",
+      cellRenderer: (params) => {
+        const type = params.data.path.split(".")[1]?.toLowerCase();
+        return (
+          <div className="flex items-center gap-2 font-medium">
+            {params.data.progress !== undefined &&
+            params.data.progress !== -1 ? (
+              <div className="flex h-6 items-center justify-center text-xs font-semibold text-muted-foreground">
+                {Math.round(params.data.progress * 100)}%
+              </div>
+            ) : (
+              <ForwardedIconComponent
+                name={FILE_ICONS[type]?.icon ?? "File"}
+                className={cn(
+                  "h-6 w-6 shrink-0",
+                  params.data.progress !== undefined
+                    ? "text-placeholder-foreground"
+                    : (FILE_ICONS[type]?.color ?? undefined),
+                )}
+              />
+            )}
+            <div
+              className={cn(
+                "flex cursor-text items-center gap-2 text-sm font-medium",
+                params.data.progress !== undefined &&
+                  params.data.progress === -1 &&
+                  "pointer-events-none text-placeholder-foreground",
+              )}
+            >
+              {params.value}.{type}
+            </div>
+            {params.data.progress !== undefined &&
+            params.data.progress === -1 ? (
+              <span className="text-[13px] text-primary">
+                Upload failed,{" "}
+                <span
+                  className="cursor-pointer text-accent-pink-foreground underline"
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    if (params.data.file) {
+                      uploadFileDirect({ file: params.data.file });
+                    }
+                  }}
+                >
+                  try again?
+                </span>
+              </span>
+            ) : (
+              <></>
+            )}
+          </div>
+        );
+      }, //This column will be twice as wide as the others
+    }, //This column will be twice as wide as the others
+    {
+      headerName: "Type",
+      field: "path",
+      flex: 1,
+      filter: "agTextColumnFilter",
+      editable: false,
+      valueFormatter: (params) => {
+        return params.value.split(".")[1]?.toUpperCase();
+      },
+      cellClass: "text-muted-foreground cursor-text select-text",
+    },
+    {
+      headerName: "Size",
+      field: "size",
+      flex: 1,
+      valueFormatter: (params) => {
+        return formatFileSize(params.value);
+      },
+      editable: false,
+      cellClass: "text-muted-foreground cursor-text select-text",
+    },
+    {
+      headerName: "Modified",
+      field: "updated_at",
+      valueFormatter: (params) => {
+        return params.data.progress
+          ? ""
+          : new Date(params.value).toLocaleString();
```

Comment:
```suggestion
          : new Date(params.value + "Z").toLocaleString();
```
This will make sure the timezone is correct by enforcing this a UTC time